src/quick-lint-js/fe/parse.cpp

```diff
@@ -328,6 +328,23 @@ void parser::error_on_sketchy_condition(expression* ast) {
   }
 }
 
+void parser::warn_on_comma_operator_in_conditional_statement(expression* ast) {
+  if (ast->kind() != expression_kind::binary_operator) return;
+
+  auto is_comma = [](string8_view s) -> bool { return s == u8","_sv; };
+
+  auto* binary_operator = static_cast<expression::binary_operator*>(ast);
+  for (span_size i = binary_operator->child_count() - 2; i >= 0; i--) {
+    source_code_span op_span = binary_operator->operator_spans_[i];
+    if (is_comma(op_span.string_view())) {
+      this->diag_reporter_->report(
+          diag_misleading_comma_operator_in_conditional_statement{.comma =
+                                                                      op_span});
+      return;
+    }
+  }
+}
```

Comment:
This looks almost identical to `warn_on_comma_operator_in_index`. We should reuse the code using a C++ template. See `consume_semicolon` for an example of what I want.

Reply:
I could not make use of the structure in the `consume_semicolon` function, because these two diagnostic functions take different arguments. Besides, their corresponding diagnostic classes have different shapes. So, instead, I put the common logic here into a separate function `find_comma_operator`.  Let me know if you would still prefer it the other way.

Reply:
> these two diagnostic functions take different arguments

Ah, I did not see that `warn_on_comma_operator_in_index` needed `left_square_span`.

> instead, I put the common logic here into a separate function `find_comma_operator`. Let me know if you would still prefer it the other way.

I prefer the way you had it before.

Reply:
Hi @strager, reverted the related changes.